engine/src/test/java/org/camunda/bpm/engine/test/bpmn/gateway/ParallelGatewayTest.java

```diff
@@ -373,4 +375,41 @@ public void testExplicitParallelGatewayAfterSignalBehavior() {
     // then
     assertEquals(3, taskService.createTaskQuery().count());
   }
+
+  @Deployment
+  @Test
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_FULL)
+  public void testParallelGatewayCancellationHistoryEvent() {
+    // given
+    // a process with one splitting and one merging parallel gateway and two parallel sequence flows between them
+    // one sequence flow has a wait state "Event_Wait", the other has none
+    var pi = runtimeService.startProcessInstanceByKey("parallelProcess");
+
+    var currentTime = new Date(5000);
+    ClockUtil.setCurrentTime(currentTime);
+
+    // when
+    // cancel "Event_Wait" first
+    // then cancel "Gateway_out" (merging parallel gateway)
+    this.runtimeService.createProcessInstanceModification(pi.getId())
+        .cancelAllForActivity("Event_Wait")
+        .cancelAllForActivity("Gateway_out")
+        .execute();
+
+    // then
+    // the whole process instance is canceled and history is produced
+    assertThat(runtimeService.createProcessInstanceQuery().processInstanceId(pi.getId()).singleResult()).isNull();
+    assertThat(historyService.createHistoricProcessInstanceQuery().processInstanceId(pi.getId()).singleResult()).isNotNull();
+
+    //
+    var historicActivityInstanceEventWait = this.historyService.createHistoricActivityInstanceQuery()
+        .activityId("Event_Wait")
+        .singleResult();
+    assertThat(historicActivityInstanceEventWait.getEndTime()).isEqualToIgnoringMillis(currentTime);
+
+    var historicActivityInstanceMergingGateway = this.historyService.createHistoricActivityInstanceQuery()
+        .activityId("Gateway_out")
+        .singleResult();
+    assertThat(historicActivityInstanceMergingGateway.getEndTime()).isEqualToIgnoringMillis(currentTime);
```

Comment:
❓ Isn't better to use `#isCloseTo()` as mentioned here: https://camunda.slack.com/archives/CL5PTD6AJ/p1699621103786889
```suggestion
    assertThat(historicActivityInstanceMergingGateway.getEndTime()).isCloseTo(currentTime, 2000);
```

Reply:
Technically, it is not necessary because the clock is fixed to a value without milliseconds. I will still change it to `isCloseTo` and non-fixed current time. Thinking about it now, I see this as the cleaner option.